### PR TITLE
Multiple features and chores

### PR DESCRIPTION
**Pull request by saszaCodes**
#### What does this PR do?
It implements multiple chores and features (it is built on top of all other unmerged branches).

Implemented chores:
- Implement Swagger
- Add password hashing

Implemented features:
- Fetch all users
- Fetch all docs for given user
- Fetch all documents
- Search users by username, email or fullname
- Search documents by title
#### How should this be manually tested?
Test all endpoints specified in API documentation (after pulling this branch, run `npm run dev` and open localhost:3000/api-docs to access easy to read documentation. Endpoints cannot be tested directly from the documentation, as this would violate CORS rules. Documentation at this point is simply a good reference for manual testing with Postman
#### Any background context you want to provide?
This PR resolves almost all cards from the Trello board, the only card left in the icebox is setting roles
#### Questions:
How to properly secure the db from SQL injection? At the moment I'm passing all user queries directly to knex, should I validate each user input, and if so, how?

### CODE REVIEW

File: src/lib/bcrypt/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable require-jsdoc */
```
Comment (Avuidrauxs) [inline src/lib/bcrypt/index.js]: Let not be doing this in files, we can add rules for them

File: src/lib/bcrypt/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import bcrypt from 'bcrypt';
+
+class Bcrypt {
+  async hashPwd(plainTextPassword) {
+    return bcrypt.hash(plainTextPassword, 10);
+  }
+
+  async comparePwd(plainTextPassword, hash) {
+    return bcrypt.compare(plainTextPassword, hash);
+  }
```
Comment (Avuidrauxs) [inline src/lib/bcrypt/index.js]: Lets have these as static methods:

```suggestion
  async static hashPwd(plainTextPassword) {
    return bcrypt.hash(plainTextPassword, 10);
  }

  async static comparePwd(plainTextPassword, hash) {
    return bcrypt.compare(plainTextPassword, hash);
  }
```

so we will use this as `Bcrypt.hashPwd('sdfs')`

File: src/lib/bcrypt/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export default new Bcrypt();
```
Comment (Avuidrauxs) [inline src/lib/bcrypt/index.js]: Based on my previous comment, lets have this like below:

```suggestion
export default Bcrypt;
```

**Review by Avuidrauxs — CHANGES_REQUESTED** [review]
Looks great, just a few comments.

**Comment by saszaCodes** [comment]
@Avuidrauxs I implemented all suggested changes, PR ready for review

**Review by Avuidrauxs — APPROVED** [review]

**Pull request merged**